### PR TITLE
extend s3 virtual host addressing rules exemptions

### PR DESCRIPTION
**Pull request by thrau**
This fixes an issue where invoking `awslocal s3 ls` would incorrectly create an s3 `ListObjects` command with the bucket name `localhost:4566` (instead of `ListBuckets`) and subsequently setting the `path` of the request object to `/localhost:4566`.

### CODE REVIEW

File: localstack/aws/protocol/parser.py [hunk 1]
```diff
@@ -980,7 +980,9 @@ def parse(self, request: HttpRequest) -> Tuple[OperationModel, Any]:
         """Handle virtual-host-addressing for S3."""
         if (
             # TODO implement a more sophisticated determination if the host contains S3 virtual host addressing
-            not request.host.startswith("s3.")
+            not request.host == "localhost"
```
Comment (alexrashed) [inline localstack/aws/protocol/parser.py:983]: When working on the ASF service router, I stumbled upon the function used for the service routing which checks if a host header indicates the S3 virtual host addressing: 
https://github.com/localstack/localstack/blob/d12d037398ec8c1db26be15ac60e481579397e36/localstack/services/s3/s3_utils.py#L90-L99

Maybe we could directly use this function here?

**Review by alexrashed — COMMENTED** [review]
Thanks for the fix! Maybe we can circumvent future issues by unifying the check (to use the same used for the service routing)?

**Comment by thrau** [comment]
updated it to use the s3 util. pretty sure it's safe to merge, just a build environment issue.

**Review by alexrashed — APPROVED** [review]
Looks great, thanks for the fix! Now we have the (quite complex) virtual host detection unified across the different components! 🚀

**Pull request merged**